src/driver-hidpp20.c

```diff
@@ -247,17 +247,57 @@ hidpp20drv_read_button(struct ratbag_button *button)
 }
 
 static void
-hidpp20drv_read_led(struct ratbag_led *led)
+hidpp20drv_read_led_1300(struct ratbag_led *led, struct hidpp20drv_data* data)
 {
-	struct ratbag_device *device = led->profile->device;
-	struct hidpp20drv_data *drv_data = ratbag_get_drv_data(device);
-	struct hidpp20_profile *profile;
-	struct hidpp20_color_led_zone_info *led_info;
-	struct hidpp20_led *h_led;
+	struct hidpp20_led_sw_ctrl_led_state state;
+	struct hidpp20_led_sw_ctrl_led_info *info;
+	int rc;
 
-	if (!(drv_data->capabilities & HIDPP_CAP_COLOR_LED_EFFECTS_8070))
+	info = &data->led_infos.leds[led->index];
+
+	rc = hidpp20_led_sw_control_get_led_state(data->dev, led->index, &state);
+
+	if (rc != 0)
 		return;
 
+	led->colordepth = RATBAG_LED_COLORDEPTH_MONOCHROME;
+
+	switch (info->type)
+	{
+	case HIDPP20_LED_TYPE_LOGO:
+		led->type = RATBAG_LED_TYPE_LOGO;
+		break;
+	case HIDPP20_LED_TYPE_BATTERY:
+	case HIDPP20_LED_TYPE_DPI:
+	case HIDPP20_LED_TYPE_PROFILE:
+	case HIDPP20_LED_TYPE_COSMETIC:
+		led->type = RATBAG_LED_TYPE_SIDE;
+		break;
+	}
+
+	switch (state.mode)
+	{
+	case HIDPP20_LED_MODE_OFF:
+		led->mode = RATBAG_LED_OFF;
+		break;
+	case HIDPP20_LED_MODE_ON:
+		led->mode = RATBAG_LED_ON;
+		break;
+	case HIDPP20_LED_MODE_BREATHING:
+		led->mode = RATBAG_LED_BREATHING;
+		led->hz = state.breathing.period;
+		led->brightness = state.breathing.max_brightness;
+		break;
+	}
```

Comment:
what happens if any other mode is set? we should default this to 'ON' imo, even when it's obviously incorrect.